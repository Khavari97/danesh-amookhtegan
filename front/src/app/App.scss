@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


