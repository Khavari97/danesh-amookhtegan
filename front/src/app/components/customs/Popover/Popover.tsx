import React from 'react';
import Popover from '@material-ui/core/Popover';
import Typography from '@material-ui/core/Typography';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        popover: {
            pointerEvents: 'none',
        },
        paper: {
            padding: theme.spacing(1),
        },
    }),
);

interface IProps {
    title: string
    children: any
}

const MouseOverPopover = (props: IProps) => {
    const classes = useStyles();
    const [anchorEl, setAnchorEl] = React.useState<HTMLElement | null>(null);

    const handlePopoverOpen = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
        if (window.innerWidth > 1024) {
            setAnchorEl(event.currentTarget);
        }
    };

    const handlePopoverClose = () => {
        if (window.innerWidth > 1024) {
            setAnchorEl(null);
        }
    };
    const open = Boolean(anchorEl);

    const children = React.Children.map(props.children, child => {
        return React.cloneElement(child, {
            onMouseEnter: handlePopoverOpen,
            onMouseLeave: handlePopoverClose
        });
    });
    return (
        <div>
            {children}
            <Popover
                id="mouse-over-popover"
                className={classes.popover}
                classes={{
                    paper: classes.paper,
                }}
                open={open}
                anchorEl={anchorEl}
                anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'left',
                }}
                transformOrigin={{
                    vertical: 'top',
                    horizontal: 'left',
                }}
                onClose={handlePopoverClose}
                disableRestoreFocus
            >
                <Typography>{props.title}</Typography>
            </Popover>
        </div>
    );
}
export default MouseOverPopover
