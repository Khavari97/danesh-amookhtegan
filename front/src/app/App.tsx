import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
