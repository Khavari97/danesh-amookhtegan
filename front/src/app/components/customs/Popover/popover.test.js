import React from "react" 
import Enzyme, { shallow,configure,mount } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import App from "./Popover";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import {Typography} from '@material-ui/core';
import Popover from '@material-ui/core/Popover';
import renderer from "react-test-renderer";
configure({ adapter: new Adapter() })

  describe('input component', () => {

     let wrapper = shallow(<App/>)
  
      it('it should have value', () => {
        wrapper.setProps({    
          title: 'string'})
          expect(wrapper.find(Typography).text()).toBe('string');
      });
      it('snapshot', () => {
        const shot = <App />
        const snapshot = renderer.create(shot).toJSON();
        expect(snapshot).toMatchSnapshot();
      })
  })