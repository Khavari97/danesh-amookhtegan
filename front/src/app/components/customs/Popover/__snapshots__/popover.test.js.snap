// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component snapshot 1`] = `<div />`;
